src/docfx/template/MustacheXrefTagParser.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class MustacheXrefTagParser
+    {
+        /// <summary>
+        /// Using `href` property to indicate xref spec resolve success.
+        /// </summary>
+        private const string XrefTagTemplate =
+        "{{#href}}" +
+        "  @resolvedTag" +
+        "{{/href}}" +
+        "{{^href}}" +
+        "  <span> {{.}} </span>" +
+        "{{/href}}";
+
+        private static readonly char[] s_trimChars = new[] { '{', ' ', '}' };
+
+        private static readonly Regex s_xrefTagMatcher = new Regex(@"<xref(.*?)\/>", RegexOptions.IgnoreCase);
+
+        public static string ProcessXrefTag(string tempalteFileName, string templateStr)
+            => s_xrefTagMatcher.Replace(templateStr, (match) => ReplaceXrefTag(match, tempalteFileName));
+
+        private static string ReplaceXrefTag(Match match, string templateFileName)
+        {
+            var xrefTag = HtmlUtility.LoadHtml(match.Value).ChildNodes.FindFirst("xref");
+            var uidName = default(string);
+            var partialName = default(string);
+            foreach (var attribute in xrefTag.Attributes)
+            {
+                if (string.Equals(attribute.Name, "uid", StringComparison.OrdinalIgnoreCase))
+                    uidName = attribute.Value.Trim(s_trimChars);
+                else if (string.Equals(attribute.Name, "template", StringComparison.OrdinalIgnoreCase))
+                    partialName = attribute.Value.Trim(s_trimChars);
+            }
+
+            var resolvedTag = partialName == null
+                ? "<a href=\"{{href}}\"> {{uid}} </a>"
+                : "{{> " + partialName + "}}";
+            var resultTemplate = XrefTagTemplate.Replace("@resolvedTag", resolvedTag);
+
+            if (uidName == null)
+            {
+                // TODO: uid may fallback to href in ProfileList
+                throw new NotSupportedException($"<xref> defined without uid in templte: {templateFileName}");
+            }
+            else if (uidName == ".")
```

Comment:
I thought it was simply:

```suggestion
            if (string.IsNullOrEmpty(uidName) || uidName == ".")
```

Throw here is probably not a good approach here as templates are some how user content?

Reply:
What about make it default `uid`?

Reply:
sure